object-construction-checker/tests/basic/Not.java

```diff
@@ -0,0 +1,61 @@
+import org.checkerframework.checker.objectconstruction.qual.*;
+
+class Not {
+
+    class Foo {
+        void a() {}
+        void b() {}
+        void c() {}
+        void notA(@CalledMethodsPredicate("!a") Foo this) {}
+        void notB(@CalledMethodsPredicate("!b") Foo this) {}
+    }
+
+    void test1(Foo f) {
+        f.notA();
+        f.notB();
+    }
+
+    void test2(Foo f) {
+        f.c();
+        f.notA();
+        f.notB();
+    }
+
+    void test3(Foo f) {
+        f.a();
+        // :: error: method.invocation.invalid
+        f.notA();
+        f.notB();
+    }
+
+    void test4(Foo f) {
+        f.b();
+        f.notA();
+        // :: error: method.invocation.invalid
+        f.notB();
+    }
+
+    void test5(Foo f) {
+        f.a();
+        f.b();
+        // :: error: method.invocation.invalid
+        f.notA();
+        // :: error: method.invocation.invalid
+        f.notB();
+    }
+
+    void callA(Foo f) {
+        f.a();
+    }
+
+    void test6(Foo f) {
+        callA(f);
+        // DEMONSTRATION OF UNSOUNDNESS
+        f.notA();
+    }
+
```

Comment:
Can we add a test case that demonstrates unsoundness due to a control-flow merge?  E.g., a case where `a()` is called the `then` branch but not the `else` branch

Reply:
Done